### PR TITLE
Prevent Sensor Data to be Requested on Remote Units when Debug Draw is enabled

### PR DESCRIPTION
**Pull request by jokoho48**
### When merged this pull request will:

1. *Title*

### CODE REVIEW

File: addons/main/functions/debug/fnc_debugDraw.sqf [hunk 1]
```diff
@@ -141,28 +141,33 @@ private _posCam = positionCameraToWorld [0, 0, 0];
...
-            drawIcon3D ["a3\ui_f\data\Map\Markers\System\dummy_ca.paa", [1, 1, 1, 1], _knowledgePosition, 1, 1, 0, "Estimated Target Position"];
-
-            if !(_lastSeen isEqualType "") then {
-                drawLine3D [_renderPos, _lastSeen, [0, 0, 1, 0.5]];
-                drawIcon3D ["a3\ui_f\data\Map\Markers\System\dummy_ca.paa", [1, 1, 1, 1], _lastSeen, 1, 1, 0, "Last Seen Position"];
+            if (!getRemoteSensorsDisabled || local _unit) then {
+                private _knowledge = _unit targetKnowledge _currentTarget;
+                private _knowledgePosition = ASLtoAGL(_knowledge select 6);
+                private _knowledgeAge = _knowledge select 2;
+                if (_knowledge select 2 == time && local _unit) then {
```
Comment (rautamiekka) [inline addons/main/functions/debug/fnc_debugDraw.sqf:148]:
```suggestion
                if (_knowledgeAge == time && local _unit) then {
```

File: addons/main/functions/debug/fnc_debugDraw.sqf [hunk 2]
```diff
@@ -141,28 +141,33 @@ private _posCam = positionCameraToWorld [0, 0, 0];
...
-                "    Position Error: ", (_knowledge select 5) toFixed 2, "</t><br/>"
-            ];
-
-            drawLine3D [_renderPos, _knowledgePosition, [0, 1, 0, 0.5]];
-            drawIcon3D ["a3\ui_f\data\Map\Markers\System\dummy_ca.paa", [1, 1, 1, 1], _knowledgePosition, 1, 1, 0, "Estimated Target Position"];
-
-            if !(_lastSeen isEqualType "") then {
-                drawLine3D [_renderPos, _lastSeen, [0, 0, 1, 0.5]];
-                drawIcon3D ["a3\ui_f\data\Map\Markers\System\dummy_ca.paa", [1, 1, 1, 1], _lastSeen, 1, 1, 0, "Last Seen Position"];
+            if (!getRemoteSensorsDisabled || local _unit) then {
```
Comment (diwako) [inline addons/main/functions/debug/fnc_debugDraw.sqf:144]:
```suggestion
            if (!getRemoteSensorsDisabled || {local _unit}) then {
```

Lazy eval

Comment (rautamiekka) [reply addons/main/functions/debug/fnc_debugDraw.sqf:144]: I'd imagine checking if an object is local is a very simple check that out-benefits lazy eval ?

Comment (diwako) [reply addons/main/functions/debug/fnc_debugDraw.sqf:144]: general code styling really. not too much needed in debug code.

Comment (jokoho48) [reply addons/main/functions/debug/fnc_debugDraw.sqf:144]: we can actually combine this and a few lines below will look into this afternoon

**Review by diwako — APPROVED** [review]

**Pull request merged**